### ISSUE TITLE
Some Expansions use the fx FSPEC mechanism for REF item prescence

### ISSUE THREAD

**Issue by victor-eh** [zoranbosnjak/asterix-specs#31]
Category 004 reserved expansion uses 2 bytes with a FX mechanism for the FSPEC instead of a fixed-size FSPEC as described here:

![image](https://github.com/user-attachments/assets/3d02ad71-dc8d-4360-9bef-da60f68eee4c)

````hs
data Expansion = Expansion
    { expCategory  :: CatNum
    , expTitle     :: Title
    , expEdition   :: Edition
    , expDate      :: Date
    , expFspecSize :: ByteSize
    , expItems     :: [Maybe (NonSpare ())]
    }

**Comment by zoranbosnjak** [comment]
Indeed, this case is not supported by the current structure. A small modification is required, something like:

```haskell
data Expansion = Expansion
    ...
    , expFspecSize :: Maybe ByteSize -- 'Nothing' represents 'fx' mechanism
    ...
```

In the  `ast` syntax, the corresponding directive will read `compound fx` instead of `compound {num}`.

Would you like to implement the category 004 ref? I would be happy to include it to the project. Until the structure change is implemented, you could validate the the rest of the spec with fixed size fspec `compound 2`.

**Comment by victor-eh** [comment]
I will be implementing the category 004 ref, though it might take a while until I have the option to contribute it here.

**Comment by zoranbosnjak** [comment]
Structure adjustment is already implemented in commit 24c25cfe2968365c53db276222925b2279a9bcb8, currently in `devel` branch. I need to run a few more tests.

**Comment by zoranbosnjak** [comment]
The structure is ready and merged to master branch. Closing issue. You are welcome to implement the actual ref when you have the option.
